shared_model/interfaces/iroha_internal/unsafe_proposal_factory.hpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_UNSAFE_PROPOSAL_FACTORY_HPP
+#define IROHA_UNSAFE_PROPOSAL_FACTORY_HPP
+
+#include <memory>
+
+#include "interfaces/common_objects/types.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    class Proposal;
+
+    /**
+     * UnsafeProposalFactory creates proposal without stateless validation
+     */
+    class UnsafeProposalFactory {
+     public:
+      virtual std::unique_ptr<Proposal> unsafeCreateProposal(
```

Comment:
The method can be made ```const```, I suppose

Reply:
SInce in abstract class method is not const, we cannot do that here. I do not want to make base class method const since it puts too much constraints on possible implementation